pkg/sources/github/github.go

```diff
@@ -919,3 +925,141 @@ func (s *Source) setProgressCompleteWithRepo(index int, offset int, repoURL stri
 
 	s.SetProgressComplete(index+offset, len(s.repos)+offset, fmt.Sprintf("Repo: %s", repoURL), encodedResumeInfo)
 }
+
+func (s *Source) scanComments(ctx context.Context, repoPath string, chunksChan chan *sources.Chunk) error {
+	s.log.Info("scanning comments")
+
+	trimmedURL := removeURLAndSplit(repoPath)
+	owner := trimmedURL[1]
+	repo := trimmedURL[2]
+
+	sortType := "created"
+	directionType := "desc"
+
+	issueOpts := &github.IssueListCommentsOptions{
+		Sort:      &sortType,
+		Direction: &directionType,
+		ListOptions: github.ListOptions{
+			PerPage: defaultPagination,
+			Page:    1,
+		},
+	}
+
+	for {
+		issueComments, _, err := s.apiClient.Issues.ListComments(ctx, owner, repo, 0, issueOpts)
+		if err != nil {
+			return err
+		}
+
+		err = s.chunkIssueComments(ctx, repo, issueComments, chunksChan, repoPath)
+		if err != nil {
+			return err
+		}
+
+		issueOpts.ListOptions.Page++
+
+		if len(issueComments) < defaultPagination {
+			break
+		}
+	}
+
+	prOpts := &github.PullRequestListCommentsOptions{
+		Sort:      sortType,
+		Direction: directionType,
+		ListOptions: github.ListOptions{
+			PerPage: defaultPagination,
+			Page:    1,
+		},
+	}
+
+	for {
+		prComments, _, err := s.apiClient.PullRequests.ListComments(ctx, owner, repo, 0, prOpts)
```

Comment:
nit: Can we replace the magic '0' w/ a const and a tiny comment.
ex: `	const defaultIssueNumber = 0 // List all comments for all issues for the repo.`

Reply:
yep for sure